### ISSUE TITLE
Screen tearing with 57 Hz + VRR

### ISSUE THREAD

**Issue by coolbho3k** [MiSTer-devel/Arcade-Cave_MiSTer#109]
Hi! I get screen tearing at the bottom of the screen with 57 Hz mode and VRR. It doesn't happen with 60 Hz mode.

Here's a brief video of the issue: https://youtu.be/C9HAeyCuBbc

The core doesn't display with my TV at 57 Hz without VRR, although it actually used to work with this display. It may be because of a TV firmware update and not the core. 

Here are my video settings:

video_mode=8
vsync_adjust=2
vrr_mode=1 (it doesn't matter if it's 1, 2, or 3)
vrr_min_framerate=0
vrr_max_framerate=120
vrr_vesa_framerate=60

The TV is a Samsung S95B.

**Comment by coolbho3k** [comment]
The Neo Geo core actually seemed fine at 57 Hz, let me try the R-Type core.

I also remember this core working without VRR at 57 Hz, with no tearing, but maybe a firmware update on the TV or MiSTer side broke it :( It's not possible to tell now.

**Comment by coolbho3k** [comment]
Yup, happens in the R-Type core too, although it's far less noticeable in that game as it's a horizontal scroller.

**Comment by desolationjones** [comment]
> I also remember this core working without VRR at 57 Hz, with no tearing, but maybe a firmware update on the TV or MiSTer side broke it :( It's not possible to tell now.

You are not alone here. I am on a Samsung Q6FN and it's been hell getting MiSTer working since about September.

**Comment by KawaiiDinosaur** [comment]
I know it's an old issue, but I fixed it by forcing AMD FreeSync on the MiSTer.ini (vrr_mode=2). I have a Samsung S90C and it recgonizes it as "FreeSync Premium", with no issues. I tested a bunch of cores with weird refresh rates and they all work without tearing.

**Comment by cholzer79** [comment]
That tearing at the lower edge of the screen is an issue that you see in gaming as well with both AMD FreeSync and NVIDIA G-Sync.
The solution there is to also enable v-sync ingame, that gets rid of that tearline near the lower edge of the screen.

https://youtu.be/YR0vNs0ZdWI?si=TjQQaFH5pIhTW-nP&t=260

However knowing that didnt realy help me yet to fully fix this problem with the Mister - only when I disable VRR I could get rid of that tearline.

I need to experiement more with this.

**Comment by MajinChibi** [comment]
> I know it's an old issue, but I fixed it by forcing AMD FreeSync on the MiSTer.ini (vrr_mode=2). I have a Samsung S90C and it recgonizes it as "FreeSync Premium", with no issues. I tested a bunch of cores with weird refresh rates and they all work without tearing.

Could you tell me, which videomode (resolution) you used? I sillt get the tearing line on my s90c

**Comment by KawaiiDinosaur** [comment]
@MajinChibi I used to use `video_mode=8` (1080p60) since my TV doesn't like 1440p for some reason. Now I use an external upscaler with my Mister outputting 1080p120, this way the TV locks at 114Hz VRR in the Cave core.

**Comment by MajinChibi** [comment]
@KawaiiDinosaur 
Thank you for answering!
When i activate vrr i get no picture. I must also activate vsync adjust. Then i get a picture and the tv shows vrr on. But then i get the tearing line, described in the first post at 50hz (which most pal games are...)
When i activate the 60hz hack in PSX core, the line is gone. But that doesn´t help for other cores :(
With 720p or 1440p (my s90c likes this resolution +.+ ) the tearing line is at another position.

Which upscaler are you using?
